.claude/rules/tests-assert-via-http-api.md

```
@@ -0,0 +1,14 @@
+---
+paths:
+  - "tests/JunimoServer.Tests/**/*.cs"
+---
+
+# E2E tests assert via the server HTTP API snapshot, not mod events
+
+E2E tests assert against the **server HTTP API snapshot** — `/cabins` (`CabinsResponse`: TotalCount/AssignedCount/AvailableCount + per-cabin OwnerId/OwnerName/IsAssigned), `/farmhands` (`FarmhandsResponse`: per-slot Id/Name/IsCustomized), `/players`, plus client-side `/status` and `/wait/*` — and the `WaitFor*` helpers on `ServerApiClient`.
+
+Mod `Diagnostics.ModEventLog.Emit(...)` events (`cabin_*`, `auth_*`, …) are transported via `Console.Out` `SDVD_EVENT ` lines → `SimpleContainerLogStreamer` → `infrastructure.jsonl` on disk. They are **diagnostics only**. No test asserts on a mod-emitted event, and there is **no "wait for mod event X" API** — the event stream is not a test assertion surface.
```

markdownlint-cli2 (0.22.1)

[warning] Spaces inside code span elements

(MD038, no-space-in-code)